core-bundle/assets/scripts/choices.js

```diff
@@ -0,0 +1,62 @@
+/*!
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+(function () {
+    let choicesList = [];
```

Comment:
In our other `MutationObserver` implementations we used a `WeakMap` instead.

https://github.com/contao/contao/blob/29cb42bdb4cfd5eccfc1a8915c271119a1e24126/core-bundle/assets/scripts/sectionwizard.js#L4-L19

But I don't know how important this would be for performance.

Reply:
Could do that - I didn't touch the JS from 2022 but there might be a slight performance increase when using a WeakMap if there is many `chosen` - aka not opposed to changing that.

Reply:
See a24a5dd20d8b4037b6639982a2c886d181e78b85